--- a/lua/entities/sg_base/modules/lib.lua	2026-10-04 17:26:37.871649900 +0200
+++ b/lua/entities/sg_base/modules/lib.lua	2026-10-04 17:22:00.845401750 +0200
@@ -1397,12 +1397,6 @@
 	return NULL;
 end
 
--- Native EAP replacement for CAP's StarGate.IsIrisClosed(gate)
-function ENT:IsIrisClosed()
-	local iris = self:GetIris();
-	return IsValid(iris) and iris.IsActivated or false;
-end
-
 -- for e2 lib by AlexALX
 function ENT:IrisToggle()
 	local iris = self:GetIris();
--- a/lua/entities/gmod_wire_expression2/core/custom/stargate.lua	2026-10-04 17:26:37.872082433 +0200
+++ b/lua/entities/gmod_wire_expression2/core/custom/stargate.lua	2026-10-04 17:22:08.748189143 +0200
@@ -537,7 +537,7 @@
 	if (this.excessPower==nil or this.excessPowerLimit==nil or not IsValid(this.overloader)) then return -1; end
 	local energyRequired = this.excessPowerLimit - this.excessPower;
 	local timeLeft = (energyRequired / this.overloader.energyPerSecond)
-	if(this:IsIrisClosed()) then
+	if(Lib.IsIrisClosed(this)) then
 		timeLeft = timeLeft * 2;
 	end
 	if (this.isOverloading) then
@@ -556,7 +556,7 @@
 	if (this.excessPower==nil or this.excessPowerLimit==nil or not IsValid(this.overloader)) then return -1; end
 	local energyRequired = this.excessPowerLimit - this.excessPower;
 	local timeLeft = (energyRequired / this.overloader.energyPerSecond)
-	if(this:IsIrisClosed()) then
+	if(Lib.IsIrisClosed(this)) then
 		timeLeft = timeLeft * 2;
 	end
 	if (this.isOverloading) then
--- a/lua/expadv/components/custom/stargate.lua	2026-10-04 17:26:37.872487026 +0200
+++ b/lua/expadv/components/custom/stargate.lua	2026-10-04 17:22:08.752573504 +0200
@@ -482,7 +482,7 @@
 	if (Entity.excessPower==nil or Entity.excessPowerLimit==nil or not IsValid(Entity.overloader)) then return -1; end
 	local energyRequired = Entity.excessPowerLimit - Entity.excessPower;
 	local timeLeft = (energyRequired / Entity.overloader.energyPerSecond)
-	if(Entity:IsIrisClosed()) then
+	if(Lib.IsIrisClosed(Entity)) then
 		timeLeft = timeLeft * 2;
 	end
 	if (Entity.isOverloading) then
@@ -502,7 +502,7 @@
 	if (Entity.excessPower==nil or Entity.excessPowerLimit==nil or not IsValid(Entity.overloader)) then return -1; end
 	local energyRequired = Entity.excessPowerLimit - Entity.excessPower;
 	local timeLeft = (energyRequired / Entity.overloader.energyPerSecond)
-	if(Entity:IsIrisClosed()) then
+	if(Lib.IsIrisClosed(Entity)) then
 		timeLeft = timeLeft * 2;
 	end
 	if (Entity.isOverloading) then
--- a/lua/wire/gates/stargate.lua	2026-10-04 17:26:37.872656972 +0200
+++ b/lua/wire/gates/stargate.lua	2026-10-04 17:22:08.755078574 +0200
@@ -233,7 +233,7 @@
 			if (Ent.excessPower==nil or Ent.excessPowerLimit==nil or not IsValid(Ent.overloader)) then return -1; end
 			local energyRequired = Ent.excessPowerLimit - Ent.excessPower;
 			local timeLeft = (energyRequired / Ent.overloader.energyPerSecond)
-			if(Ent:IsIrisClosed()) then
+			if(Lib.IsIrisClosed(Ent)) then
 				timeLeft = timeLeft * 2;
 			end
 			if (Ent.isOverloading) then
--- a/lua/starfall/libs_sv/stargate.lua	2026-10-04 17:26:37.872290335 +0200
+++ b/lua/starfall/libs_sv/stargate.lua	2026-10-04 17:22:08.749477743 +0200
@@ -224,7 +224,7 @@
 	if (this.excessPower==nil or this.excessPowerLimit==nil or not IsValid(this.overloader)) then return false; end
 	local energyRequired = this.excessPowerLimit - this.excessPower;
 	local timeLeft = (energyRequired / this.overloader.energyPerSecond)
-	if(this:IsIrisClosed()) then
+	if(Lib.IsIrisClosed(this)) then
 		timeLeft = timeLeft * 2;
 	end
 	if (this.isOverloading) then
@@ -725,7 +725,7 @@
 	if (this.excessPower==nil or this.excessPowerLimit==nil or not IsValid(this.overloader)) then return false; end
 	local energyRequired = this.excessPowerLimit - this.excessPower;
 	local timeLeft = (energyRequired / this.overloader.energyPerSecond)
-	if(this:IsIrisClosed()) then
+	if(Lib.IsIrisClosed(this)) then
 		timeLeft = timeLeft * 2;
 	end
 	if (this.isOverloading) then
